packages/pwa-kit-react-sdk/src/ssr/universal/components/with-react-query/index.test.js

```diff
@@ -29,6 +39,49 @@ describe('withReactQuery', function () {
         expect(screen.getByText(/Hello world/i)).toBeInTheDocument()
     })
 
+    test('`beforeHydrate` called on mount', () => {
```

Comment:
We have tests for the `beforeHydrate`. Can we also have a test on the retail-react-app too? 

Say for a page like PLP, have a test to verify that there is no skeleton and no productSearch refetch. Otherwise, if there's a regression in this behaviour, we can only detect it manually.

Reply:
So we currently have _unit_ test for our a page components, and how those are done is purely run in a client-side manner (e.g. the server side data fetching and rendering is not built into the unit tests). We do have E2E tests that would probably be a better place to home the requests test. Unfortunately all our E2E tests don't access the PLP directly, they simply do things like guest flow and registered flow, meaning the PLP is always a soft navigation.

How married are you to having a one off e2e test to check that skeletons don't show up. It's not a bad idea, but I know we have been talking about having a separate push for uplifting out e2e tests and that might be a goos AC for that work.

Reply:
I've created a [ticket](https://gus.lightning.force.com/lightning/_classic/%2Fa07EE00001pKG9LYAW) to create e2e on a broader scope for testing hydration state across the template.